### PR TITLE
Document RISC-V following Itanium C++ ABI for C++ mangling rule.

### PR DESCRIPTION
**Pull request by kito-cheng**
Address #175.

### CODE REVIEW

File: riscv-elf.md [hunk 1]
```diff
@@ -447,6 +448,14 @@ of the vararg save area.  The `va_arg` macro will increment its `va_list`
 argument according to the size of the given type, taking into account the
 rules about 2&times;XLEN aligned arguments being passed in "aligned" register pairs.
 
+## <a name=c++-mangling></a>C++ Mangling for types
+
+C++ Mangling for RISC-V are following [Itanium C++ ABI](http://itanium-cxx-abi.github.io/cxx-abi/), there are no RISC-V
```
Comment (jim-wilson) [inline riscv-elf.md:453]:
```suggestion
C++ Mangling for RISC-V is following the [Itanium C++ ABI](http://itanium-cxx-abi.github.io/cxx-abi/), there are no RISC-V
```

File: riscv-elf.md [hunk 2]
```diff
@@ -447,6 +448,14 @@ of the vararg save area.  The `va_arg` macro will increment its `va_list`
...
+specific mangling rule.
```
Comment (jim-wilson) [inline riscv-elf.md:454]:
```suggestion
specific mangling rules.
```

File: riscv-elf.md [hunk 3]
```diff
@@ -447,6 +448,14 @@ of the vararg save area.  The `va_arg` macro will increment its `va_list`
...
+
+See [Type encodings](http://itanium-cxx-abi.github.io/cxx-abi/abi.html#mangling-type)
+for more detail of how to mangling a type.
```
Comment (jim-wilson) [inline riscv-elf.md:457]:
```suggestion
for more detail of how to mangle a type.
```

**Review by jim-wilson — APPROVED** [review]
Optionally change :C++ Mangling" to "C++ Name Mangling" in the 3 places it occurs unless maybe people don't use the longer phrase anymore.

**Comment by kito-cheng** [comment]
Changes:
- Apply Jim's suggestion.

File: riscv-elf.md [hunk 4]
```diff
@@ -447,6 +448,14 @@ of the vararg save area.  The `va_arg` macro will increment its `va_list`
 argument according to the size of the given type, taking into account the
 rules about 2&times;XLEN aligned arguments being passed in "aligned" register pairs.
 
+## <a name=c++-name-mangling></a>C++ Name Mangling
+
+C++ name mangling for RISC-V is following the [Itanium C++ ABI](http://itanium-cxx-abi.github.io/cxx-abi/), there are no RISC-V
```
Comment (jrtc27) [inline riscv-elf.md:453]:
```suggestion
C++ name mangling for RISC-V follows the [Itanium C++ ABI](http://itanium-cxx-abi.github.io/cxx-abi/); there are no RISC-V
```

File: riscv-elf.md [hunk 5]
```diff
@@ -447,6 +448,14 @@ of the vararg save area.  The `va_arg` macro will increment its `va_list`
...
+specific mangling rules.
+
+See [Type encodings](http://itanium-cxx-abi.github.io/cxx-abi/abi.html#mangling-type)
+for more detail of how to mangle a type.
```
Comment (jrtc27) [inline riscv-elf.md:457]:
```suggestion
for more detail on how to mangle types.
```

**Comment by kito-cheng** [comment]
Changes:
- Apply Jessica's suggestion.

**Comment by kito-cheng** [comment]
Changes:
- Rebase to master.

**Comment by kito-cheng** [comment]
@jrtc27 let me know if OK to you :)

File: riscv-elf.md [hunk 6]
```diff
@@ -14,10 +14,11 @@
 	* [Named ABIs](#named-abis)
 	* [Default ABIs](#default-abis)
 	* [Code models](#code-models)
-3. [C type details](#c-types)
-	* [C type sizes and alignments](#c-type-sizes)
-	* [C type representations](#c-type-representation)
+3. [C/C++ type details](#c-types)
+	* [C/c++ type sizes and alignments](#c-type-sizes)
+	* [C/c++ type representations](#c-type-representation)
```
Comment (jrtc27) [inline riscv-elf.md:19]: Capital C for C++

**Comment by jrtc27** [comment]
Either you never applied my suggestions or you lost them in a rebase

File: riscv-elf.md (hunk 6 above)
Comment (kito-cheng) [reply riscv-elf.md]: Fixed.

**Comment by kito-cheng** [comment]
> Either you never applied my suggestions or you lost them in a rebase

I guess those changes might lost during rebase...updated and included those suggestion now.

**Review by jrtc27 — APPROVED** [review]

**Pull request merged**